rust/agama-server/src/questions/web.rs

```diff
@@ -0,0 +1,151 @@
+//! This module implements the web API for the software module.
+//!
+//! The module offers two public functions:
+//!
+//! * `questions_service` which returns the Axum service.
+//! * `questions_stream` which offers an stream that emits questions related signals.
+
+use std::collections::HashMap;
+use crate::{error::Error, web::Event};
+use agama_lib::{
+    error::ServiceError, proxies::Questions1Proxy,
+};
+use axum::{
+    extract::{State, Path},
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    routing::{get, put},
+    Json, Router,
+};
+use tokio_stream::{Stream, StreamExt};
+use zbus::fdo::ObjectManagerProxy;
+use thiserror::Error;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+// TODO: move to lib
```

Comment:
:+1:

Reply:
only tricky part with moving to lib is that structs for web does not live there...There are GenericQuestion struct, so after move I will need to implement some `Into`

Reply:
Well, actually `GenericQuestion` can be reused in other contexts, right? So I do not see the problem of moving that struct to lib too.